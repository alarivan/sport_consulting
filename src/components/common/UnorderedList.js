import React from "react";
import styled from "styled-components";

const UL = styled.ul`
  font-size: 1.2rem;

  & > li {
    margin-bottom: 2rem;
  }
`;

export default props => (
  <UL>
    {props.items.map((item, index) => (
      <li key={index}>{item}</li>
    ))}
  </UL>
);
