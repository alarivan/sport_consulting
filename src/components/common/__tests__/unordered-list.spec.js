import React from "react"
import { render } from "@testing-library/react"
import UnorderedList from "../UnorderedList.js"

describe("UnorderedList", () => {
  it("renders correctly without items", () => {
    const { asFragment } = render(<UnorderedList items={[]} />)

    expect(asFragment()).toMatchSnapshot()
  })

  it("renders correctly with items", () => {
    const { asFragment } = render(
      <UnorderedList items={["item 1", "item 2"]} />
    )

    expect(asFragment()).toMatchSnapshot()
  })
})
