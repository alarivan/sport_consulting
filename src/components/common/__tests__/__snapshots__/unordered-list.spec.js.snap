// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UnorderedList renders correctly with items 1`] = `
<DocumentFragment>
  .c0 {
  font-size: 1.2rem;
}

.c0 > li {
  margin-bottom: 2rem;
}

<ul
    class="c0"
  >
    <li>
      item 1
    </li>
    <li>
      item 2
    </li>
  </ul>
</DocumentFragment>
`;

exports[`UnorderedList renders correctly without items 1`] = `
<DocumentFragment>
  .c0 {
  font-size: 1.2rem;
}

.c0 > li {
  margin-bottom: 2rem;
}

<ul
    class="c0"
  />
</DocumentFragment>
`;
